apps/ejabberd/src/mongoose_metrics.erl

```diff
@@ -276,9 +276,16 @@ create_metrics(Host) ->
     lists:foreach(fun(Name) -> ensure_metric(Name, counter) end,
                   get_total_counters(Host)).
 
-ensure_metric(Metric, Type) when is_list(Metric) ->
+ensure_metric(Metric, Type) when is_tuple(Type) ->
```

Comment:
Is this `Type` tuple well defined? I mean, do we know if it's always of form `{Type, Sth}`. If so then, I think it would be better to place this check as pattern matching in function's head.

```erlang
ensure_metric(Metric, {El1, _El2} = Type) ->
    ensure_metric(Metric, Type, El1);
```

Of course `El1` should have more meaningful name.